Tools/SwiftBrowser/Source/ContentView.swift

```diff
@@ -55,7 +55,97 @@ fileprivate struct ToolbarBackForwardMenuView: View {
     }
 }
 
-fileprivate struct DialogActionsView: View {
+private struct MediaCaptureStateButtonView: View {
+    struct LabelConfiguration {
+        let activeSystemImage: String
+        let mutedSystemImage: String
+    }
+
+    @Binding var captureState: WKMediaCaptureState
+
+    let configuration: LabelConfiguration
+
+    var body: some View {
+        switch captureState {
+        case .none:
+            EmptyView()
+
+        case .active:
+            Button {
+                captureState = .muted
+            } label: {
+                Label("Mute", systemImage: configuration.activeSystemImage)
+                    .labelStyle(.iconOnly)
+            }
+
+        case .muted:
+            Button {
```

Comment:
If this were for real, I’d suggest some fun stuff you can do with SF Symbols to make this a nicer state transition, but for a test app this is fine :P

Reply:
you've piqued my curiosity now 😛